src/lib/swapper/swappers/LifiSwapper/utils/transformLifiFeeData/transformLifiFeeData.ts

```diff
@@ -27,54 +30,45 @@ export const transformLifiFeeData = ({
-  const allRouteFeeCosts = selectedRoute.steps.flatMap(step => step.estimate.feeCosts ?? [])
-
-  const buyAssetAddress = getEvmAssetAddress(buyAsset)
-
-  const buyAssetRouteFeeCosts = allRouteFeeCosts.filter(
-    feeCost => feeCost.token.address === buyAssetAddress,
-  )
-
-  // all fees that are not the buy asset (there may be multiple different tokens)
-  const sellAssetRouteFeeCosts = allRouteFeeCosts.filter(
-    feeCost => feeCost.token.address !== buyAssetAddress,
-  )
-
-  // this is the sum of all `feeCost` against the buy asset in USD
-  // need to manually convert them to USD because lifi rounds to the nearest dollar
-  const buyAssetTradeFeeUsd =
-    buyAssetRouteFeeCosts
-      .map(feeCost =>
-        baseUnitToHuman({
-          value: feeCost.amount,
-          inputExponent: feeCost.token.decimals,
-        }).multipliedBy(bnOrZero(feeCost.token.priceUSD)),
-      )
-      .reduce((acc, amountUsd) => acc.plus(amountUsd), bn(0)) ?? bn(0)
-
-  // this is the sum of all `feeCost` against the sell asset in USD
-  // need to manually convert them to USD because lifi rounds to the nearest dollar
-  const initialSellAssetTradeFeeUsd =
-    sellAssetRouteFeeCosts
-      .map(feeCost =>
-        baseUnitToPrecision({
-          value: feeCost.amount,
-          inputExponent: feeCost.token.decimals,
-        }).multipliedBy(bnOrZero(feeCost.token.priceUSD)),
-      )
-      .reduce((acc, amountUsd) => acc.plus(amountUsd), bn(0)) ?? bn(0)
+  const allRouteFeeCosts = selectedRoute.steps
+    .flatMap(step => step.estimate.feeCosts ?? [])
+    .filter(feeCost => !(feeCost as { included?: boolean }).included) // filter out included fees
 
   const feeAsset = getFeeAssets(chainId)
 
-  const { networkFeeCryptoBaseUnit, sellAssetTradeFeeUsd } = processGasCosts({
+  const { networkFeeCryptoBaseUnit, otherGasCosts } = processGasCosts({
     feeAsset,
     allRouteGasCosts,
-    initialSellAssetTradeFeeUsd,
   })
 
+  const assets = selectAssets(store.getState())
+
+  // TEMP: jumble in gas costs not denominated in the sell chain fee asset with protocol fees
+  // until our UI can convey the concept of gas fees denominated in multiple tokens
+  const protocolFees = [...allRouteFeeCosts, ...otherGasCosts].reduce((acc, feeCost) => {
+    const { amount, token } = feeCost
+    const assetId = lifiTokenToAssetId(token)
+    const asset = assets[assetId]
+    if (acc[assetId] === undefined) {
+      acc[assetId] = {
+        amountCryptoBaseUnit: amount,
+        asset: {
+          chainId: asset?.chainId ?? lifiChainIdToChainId(token.chainId),
+          precision: asset?.precision ?? token.decimals,
+          symbol: asset?.symbol ?? token.symbol,
+        },
+        requiresBalance: true,
+      }
+    } else {
+      acc[assetId].amountCryptoBaseUnit = bn(acc[assetId].amountCryptoBaseUnit)
+        .plus(amount)
+        .toString()
+    }
+    return acc
+  }, {} as Record<AssetId, ProtocolFee>)
```

Comment:
can we put this as a type assertion in the `reduce<Record<AssetId, ProtocolFee>>(...)` rather than cast?

Reply:
yeah nice will do